### PR TITLE
Shmfs bug fix

### PR DESCRIPTION
**Pull request by Zhangshoukui**
## Summary
Repair partial interface
## Impact
shmfs
## Testing
local shmfs test

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, the PR does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:**
    * Lacks a clear explanation of why the change is necessary. What was wrong with the partial interface? 
    * No details on how the change works. What specific modifications were made to repair the interface?
* **Impact:**
    * While "shmfs" is mentioned, it's unclear what the specific impact is. Does this fix a bug, add a feature, or improve performance?
    *  The impact sections are largely left blank. Address all the points (user impact, build impact, etc.) with either "NO" or a "YES" followed by a description.
* **Testing:**
    *  "local shmfs test" is too vague. Provide:
        * Specific host OS and compiler details.
        * Target architecture and board configuration.
        * Relevant snippets of the actual testing logs (before and after).

**To improve this PR:**

1. **Expand the Summary:** Clearly describe the problem, the solution, and any related issues/PRs.
2. **Complete the Impact section:**  Address all impact areas with either "NO" or a "YES" and a brief explanation.
3. **Provide Detailed Testing Information:** Specify your build and target environments and include relevant log snippets showcasing the change's effect.

File: fs/shm/shmfs_alloc.c [hunk 1]
```diff
@@ -148,20 +148,16 @@ FAR struct shmfs_object_s *shmfs_alloc_object(size_t length)
...
-
   if (object)
     {
 #if defined(CONFIG_BUILD_FLAT)
       fs_heap_free(object->paddr);
 #elif defined(CONFIG_BUILD_PROTECTED)
       kumm_free(object->paddr);
 #elif defined(CONFIG_BUILD_KERNEL)
-      pages = &object->paddr;
+      size_t i;
```
Comment (xiaoxiang781216) [inline fs/shm/shmfs_alloc.c:158]: move before line 153

Comment (xiaoxiang781216) [reply fs/shm/shmfs_alloc.c:159]: ALL variables need be declared at the beginning of block to confirm c89 spec.

File: fs/shm/shmfs.c [hunk 2]
```diff
@@ -387,11 +387,19 @@ static int shmfs_unmap_area(FAR struct task_group_s *group,
...
+        }
 
-      /* Add the virtual memory back to the shared memory pool */
+      /* Free the virtual address space */
 
       vm_release_region(get_group_mm(group), vaddr, length);
     }
+  else
+    {
+      return -EINVAL;
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply fs/shm/shmfs.c:401]: no change

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**